test/suites/edgexfoundry/proxy_test.go

```diff
@@ -123,3 +134,42 @@ func generateCerts(tmpDir string) (string, string) {
 
 	return caCertFile, caKeyFile
 }
+
+// waitCertInstall checks if certificate is fully installed
+// up to a maximum retry number
+func waitCertInstall(t *testing.T, caKey string, maxRetry int) {
+
+	type ResponseBody struct {
+		Data []struct {
+			Key string
+		}
+	}
+
+	res := &ResponseBody{}
+
+	for i := 1; ; i++ {
+		t.Logf("Checking certificate installation. Retry %d/%d", i, maxRetry)
+
+		client := &http.Client{}
+		req, err := http.NewRequest("GET", "http://localhost:8001/certificates", nil)
+		require.NoError(t, err)
+		resp, err := client.Do(req)
+		require.NoError(t, err)
+		defer resp.Body.Close()
+
+		data, err := io.ReadAll(resp.Body)
+		err = json.Unmarshal(data, res)
+		require.NoError(t, err)
+
+		certLength := len(res.Data)
+		if i == maxRetry && certLength == 0 {
```

Comment:
`len(res.Data)` is not the length of certificate. It is the total number of certificateS.

Reply:
Updated. It is called `certsNumber` now. 

I was intent to call it certsLength (The length of certs)